device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/CorrelatingMessageCallback.java

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.device;
+
+import com.microsoft.azure.sdk.iot.device.transport.IotHubTransportPacket;
+
+/**
+ * This interface has a number of call backs to notify when a message is in a specific part of the {@code IotHubTransport} lifecycle.
```

Comment:
Who implements it? SDK or customers?

Reply:
Customers. Added a note that there is no default implemenation